lib/plugins/aws/deploy/lib/configureStack.js

```diff
@@ -4,106 +4,163 @@ const _ = require('lodash');
-        path.join(this.serverless.config.serverlessPath,
-          'plugins',
-          'aws',
-          'deploy',
-          'lib',
-          'iam-policy-lambda-execution-template.json')
-      );
+  // merge in the iamPolicyLambdaTemplate
+  const iamPolicyLambdaExecutionTemplate = loadPolicyTemplate.call(this);
 
-      // set the necessary variables for the IamPolicyLambda
-      iamPolicyLambdaExecutionTemplate
-        .IamPolicyLambdaExecution
-        .Properties
-        .PolicyName = `${this.options.stage}-${this.serverless.service.service}-lambda`;
+  // set the necessary variables for the IamPolicyLambda
+  iamPolicyLambdaExecutionTemplate
+    .IamPolicyLambdaExecution
+    .Properties
+    .PolicyName = `${this.serverless.config.stage}-${this.serverless.service.service}-lambda`;
 
-      iamPolicyLambdaExecutionTemplate
-        .IamPolicyLambdaExecution
-        .Properties
-        .PolicyDocument
-        .Statement[0]
-        .Resource = `arn:aws:logs:${this.options.region}:*:*`;
+  iamPolicyLambdaExecutionTemplate
+    .IamPolicyLambdaExecution
+    .Properties
+    .PolicyDocument
+    .Statement[0]
+    .Resource = `arn:aws:logs:${this.serverless.config.region}:*:*`;
 
-      _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
-        iamPolicyLambdaExecutionTemplate);
+  _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
+    iamPolicyLambdaExecutionTemplate);
 
+  // add custom iam role statements
+  if (this.serverless.service.provider.iamRoleStatements &&
+    this.serverless.service.provider.iamRoleStatements instanceof Array) {
+    this.serverless.service.provider.compiledCloudFormationTemplate
+      .Resources
+      .IamPolicyLambdaExecution
+      .Properties
+      .PolicyDocument
+      .Statement = this.serverless.service.provider.compiledCloudFormationTemplate
+      .Resources
+      .IamPolicyLambdaExecution
+      .Properties
+      .PolicyDocument
+      .Statement.concat(this.serverless.service.provider.iamRoleStatements);
+  }
+}
+/**
+ * Add a defined deployment bucket after validating it.
+ * @returns {*} A promise that will be resolved on completion
+ */
+function addDeploymentBucket() {
+  const bucketName = this.serverless.service.provider.deploymentBucket;
 
-      // add custom iam role statements
-      if (this.serverless.service.provider.iamRoleStatements &&
-        this.serverless.service.provider.iamRoleStatements instanceof Array) {
+  if (bucketName) {
+    return BbPromise.bind(this)
+      .then(() => this.validateS3BucketName(bucketName))
+      .then(() => this.sdk.request('S3',
+        'getBucketLocation',
+        {
+          Bucket: bucketName,
+        },
+        this.serverless.config.stage,
+        this.serverless.config.region
+      ))
+      .then(result => {
+        if (result.LocationConstraint !== this.serverless.config.region) {
```

Comment:
We can't use `this.serverless.config.region` as it might be unset, e.g. when running `serverless deploy --noDeploy` you get an undefined in the CF template.